File: client/src/app/services/match-context/match-context.service.ts
import { Injectable } from '@angular/core';
import { MatchContext } from '@app/constants/states';

@Injectable({
    providedIn: 'root',
})
export class MatchContextService {
    private context: MatchContext;

    resetContext(): void {
        this.context = MatchContext.Null;
    }

    setContext(context: MatchContext): void {
        this.context = context;
    }

    getContext(): MatchContext {
        return this.context;
    }
}

File: client-mobile/app/src/main/java/com/example/polyquiz/match/domain/MatchContextService.kt
package com.example.polyquiz.match.domain
import com.example.polyquiz.constants.MatchContext
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow

object MatchContextService {
    private val _context = MutableStateFlow(MatchContext.Null)
    val context = _context.asStateFlow()
    fun resetContext() {
        _context.value = MatchContext.Null
    }

    fun setContext(context: MatchContext) {
        _context.value = context
    }

    fun getContext(): MatchContext {
        return context.value
    }
}

